docs/Course Wiki/ECE Course Offerings/ECE120.md

```diff
@@ -0,0 +1,55 @@
+
+## Course Structure
+
+The class starts from the very basics of computer engineering. The initial two to three weeks stress abstraction of the computer, bit arithmetic, Boolean logic, and base conversion. This then builds up to a week of an introduction to the C programming language. This week covers input/output, conditional statements, and loops. Afterward, the course material introduces combinational logic implementations, greater Boolean implementations, and Karnaugh maps. The following month is where the course contains its most important applicable content by going through utilizing combination and sequential logic for understanding the functionalities of these structures. Importance is placed on understanding the implementations of using such structures in memory constructs and designing finite state machines. The final quarter of the class teaches students LC-3: a simplified assembly language designed to introduce control path analysis and low-level coding. The LC-3 is explained completely by using the concepts that have led up to that point in the class. 
+
+The course has weekly homeworks that are graded based on completion rather than correctness, which encourages students to at least attempt the homeworks and try their best on every question. There are then corrections submitted for questions marked wrong by graders, which make up the remaining points for the homework category. the course also has weekly labs, which are generally asynchronous programming labs in C and later LC-3 assembly, however there are also 3 breadboarding labs where students will use digital TTL chips to create digital circuits on a breadboard. Weekly discussion sections led by TAs are held every Thursday, in which students work together on a worksheet of problems to help better their understanding of topics covered that week in class. Three midterm exams and a cumulative final conclude the examination of the course. There are not very many practice exams on offer, so it is recommended to keep up with the homework for each week in order to maintain a proper understanding of the covered concepts.
+
+There is an additional "Honors Lab" students can take to gain James Scholar credit, newly officially listed this semester (Fall 2023) as [ECE198HL]. This course allows students to design and build any project they want (with TA approval). The department will fund any project in this section (within reason). James Scholars are highly encouraged to take this course to obtain honors credit, but even non-James Scholars may take it to gain more project-based experience early on in their academic careers.
```

Comment:
Add the markdown link to ECE198HL.

Reply:
Is that link up yet? Didn't realize it was, I'll add it in.